### PR TITLE
[W5.7][W17-1] Jiang Hengyuan

### PR DESCRIPTION
**Pull request by jhengy**
work in progress

### CODE REVIEW

**Comment by jhengy** [comment]
ready for review

**Review by anubh-v — COMMENTED** [review]
Just wondering, what's the standard you are using for commit messages? Is there something to follow?
Personally, I try to have a distinct title for each message by having a line break after my first sentence.

**Comment by anubh-v** [comment]
I also did the same enhancement by the way - do take a look if possible
I used a different method, where I tried to maintain the abstraction of the ListCommand.

**Comment by anubh-v** [comment]
I think we should not include the commits from existing Pull Requests (such as W4.11e)
Not too sure on this.

File: src/seedu/addressbook/Main.java [hunk 1]
```diff
@@ -108,7 +108,7 @@ private void recordResult(CommandResult result) {
     private CommandResult executeCommand(Command command)  {
         try {
             command.setData(addressBook, lastShownList);
-            CommandResult result = command.execute();
+            CommandResult result = command.execute(); // command Result contains a String which is feedback to the user and a list of relevant persons
```
Comment (amrut-prabhu) [inline src/seedu/addressbook/Main.java:111]: Is this comment relevant here?
Should it be in the `@return` JavaDocs for the `execute()` method instead?

File: src/seedu/addressbook/commands/AddCommand.java [hunk 2]
```diff
@@ -1,18 +1,13 @@
...
-
 import seedu.addressbook.data.exception.IllegalValueException;
-import seedu.addressbook.data.person.Address;
-import seedu.addressbook.data.person.Email;
-import seedu.addressbook.data.person.Name;
-import seedu.addressbook.data.person.Person;
-import seedu.addressbook.data.person.Phone;
-import seedu.addressbook.data.person.ReadOnlyPerson;
-import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.person.*;
```
Comment (amrut-prabhu) [inline src/seedu/addressbook/commands/AddCommand.java:4]: Avoid batch imports.

Tip: You can change the IntelliJ settings to help do this.

File: test/expected.txt [hunk 3]
```diff
@@ -86,6 +86,8 @@
 || Tags names should be alphanumeric
 || ===================================================
 || Enter command: || [Command entered:  add Adam Brown p/111111 e/adam@gmail.com a/111, alpha street]
+|| 	1. Adam Brown Phone: 111111 Email: adam@gmail.com Address: 111, alpha street Tags:
```
Comment (amrut-prabhu) [inline test/expected.txt:89]: Good job for updating I/O tests!

File: test/java/seedu/addressbook/commands/AddCommandTest.java [hunk 4]
```diff
@@ -1,29 +1,15 @@
...
-import seedu.addressbook.data.person.Person;
-import seedu.addressbook.data.person.Phone;
-import seedu.addressbook.data.person.ReadOnlyPerson;
-import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.person.*;
 import seedu.addressbook.util.TestUtil;
 
+import java.util.*;
+
+import static org.junit.Assert.*;
```
Comment (amrut-prabhu) [inline test/java/seedu/addressbook/commands/AddCommandTest.java:11]: As mentioned above, avoid batch imports

File: test/java/seedu/addressbook/commands/AddCommandTest.java [hunk 5]
```diff
@@ -125,7 +111,8 @@ public void addCommand_emptyAddressBook_addressBookContainsPerson() {
 
         assertTrue(people.contains(p));
         assertEquals(1, people.immutableListView().size());
-        assertFalse(result.getRelevantPersons().isPresent());
+        //assertFalse(result.getRelevantPersons().isPresent());// result now have a component of relevant person
```
Comment (amrut-prabhu) [inline test/java/seedu/addressbook/commands/AddCommandTest.java:114]: Good job for updating JUnit tests!

On a side note, instead of commenting out unused code, delete it. We have RCS in case we need to access it again.

File: test/java/seedu/addressbook/common/UtilsTest.java [hunk 6]
```diff
@@ -43,4 +42,41 @@ private void assertAreUnique(Object... objects) {
     private void assertNotUnique(Object... objects) {
         assertFalse(Utils.elementsAreUnique(Arrays.asList(objects)));
     }
+
+
+    // this is a method for testing the method Utils.isAnyNull
+    @Test 
+    public void isAnyNullTest() throws Exception {
```
Comment (amrut-prabhu) [inline test/java/seedu/addressbook/common/UtilsTest.java:49]: You should make clean PRs!
Do not have unrelated code changes.

File: test/java/seedu/addressbook/commands/AddCommandTest.java [hunk 7]
```diff
@@ -125,7 +111,8 @@ public void addCommand_emptyAddressBook_addressBookContainsPerson() {
...
+        assertTrue(result.getRelevantPersons().isPresent());
```
Comment (amrut-prabhu) [inline test/java/seedu/addressbook/commands/AddCommandTest.java:115]: Would it be better to check what persons are present as well?

**Review by amrut-prabhu — APPROVED** [review]
Good job @jhengy 
Remember to update User Guide as well.
Also, I understand that you added comments in the code to better understand it yourself. But, it's better if you rely on the JavaDocs of the methods or add more concise and relevant comments so that the codebase doesn't look cluttered.

Also, in the future, remember to ensure that it is a clean PR! And remember to update the `work in progress` comment.

Please close the PR after reading the comments.
